Rework the Kafka Guide (#124)
crate/cratedb-guide: docs/integrate/etl/kafka-connect.md

(kafka-connect)=
# Data Ingestion Pipeline with Kafka and CrateDB

This guide describes a dockerized procedure for integrating CrateDB with Kafka
Connect. By following these steps, you will set up a pipeline to ingest data 

> from Kafka topics into CrateDB seamlessly.

Reviewer: Active voice, very good.

that can stream data to and from Kafka.

Thanks to their compatibility [Kafka Connect JDBC connector] with the [
PostgreSQL driver] allow to designate CrateDB as a sink target, with the 
following example connector definition:

```json
{
  "name": "crate-jdbc-sink",
  "config": {
    "connector.class": "io.confluent.connect.jdbc.JdbcSinkConnector",
    "tasks.max": "1",
    "topics": "sensor-data-topic",
    "connection.url": "jdbc:postgresql://cratedb:5432/crate?user=crate&sslmode=disable",
    "dialect.name": "PostgreSqlDatabaseDialect",
    "auto.create": "true",
    "auto.evolve": "true",
    "insert.mode": "insert",
    "pk.mode": "none",
    "value.converter": "io.confluent.connect.avro.AvroConverter",
    "value.converter.schemas.enable": "true",
    "value.converter.schema.registry.url": "http://schema-registry:8081",
    "key.converter": "org.apache.kafka.connect.storage.StringConverter",
    "key.converter.schemas.enable": "false",
    "table.name.format": "sensortable",
    "schema.pattern": "doc"
  }
}
```

This results in the following architecture:

![image](/_assets/img/integrations/kafka-connect.png){width=650px}

## Setup

> To illustrate how this architecture can be used, we will create a scenario 
> where we have machine sensor data from a series of weather stations being 
> ingested into a Kafka topic. This data could be used in a reactive sense: for 

Reviewer: Active voice?
```suggestion
To illustrate how this architecture can be used, let's create a scenario 
where machine sensor data from a series of weather stations is 
ingested into a Kafka topic. This data could be used in a reactive sense: for 
```

example, a micro-controller could consume from this topic to turn on air 

> conditioning if the temperature were to rise above a certain threshold. Aside
> this use of the data, we want to ingest it into CrateDB. This allows 
> us to do long term data analytics, like predicting the weather trends for 
> example. Payload from each sensor may look like this:

Reviewer: ```suggestion
conditioning if the temperature were to rise above a certain threshold. Aside
this use of the data, it should be stored into CrateDB. This allows 
to do long term data analytics, like predicting the weather trends for 
example. Payload from each sensor may look like this:
```
